### PR TITLE
disable broken macs that use rosetta, add macos-06 and update macos instructions

### PR DESCRIPTION
**Pull request by steveej**
- [x] deployed to macos-06 and linux-builder-01

### CODE REVIEW

File: modules/nixos/nix-build-distributor.nix [hunk 1]
```diff
@@ -15,7 +15,8 @@
       sshUser = "builder";
       protocol = "ssh-ng";
       system = "x86_64-darwin";
-      maxJobs = 3;
+      # has a broken nix installation
```
Comment (jost-s) [inline modules/nixos/nix-build-distributor.nix:18]: Can you not fix the installation or re-install?

File: modules/nixos/nix-build-distributor.nix [hunk 2]
```diff
@@ -53,9 +54,9 @@
       hostName = "167.235.38.111";
       sshUser = "builder";
       protocol = "ssh-ng";
-      system = "x86_64-darwin";
+      system = "aarch64-darwin";
```
Comment (jost-s) [inline modules/nixos/nix-build-distributor.nix:57]: Does this mean it's running on an arm64 machine and the x64-darwin is virtualized?

**Review by jost-s — APPROVED** [review]

File: modules/nixos/nix-build-distributor.nix (hunk 1 above)
Comment (steveej) [reply modules/nixos/nix-build-distributor.nix:18]: couldn't fix it or reinstall it requested a machine wipe but for now it's what it is.

File: modules/nixos/nix-build-distributor.nix (hunk 2 above)
Comment (steveej) [reply modules/nixos/nix-build-distributor.nix:57]: yes, that was the case prior to this.

**Pull request merged**